packages/components/src/tooltip/test/index.js

```diff
@@ -1,371 +1,357 @@
-			} );
+			const user = userEvent.setup();
 
-			const { container } = render(
-				<Tooltip text="Show helpful text here">
-					<button disabled>Click me</button>
+			render(
+				<Tooltip { ...props }>
+					<Button disabled>Click me</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button', { name: 'Click me' } );
+			const button = screen.getByRole( 'button', { name: /Click me/i } );
+
 			expect( button ).toBeVisible();
 			expect( button ).toBeDisabled();
 
-			// Note: this is testing for implementation details,
-			// but couldn't find a better way.
-			const buttonRect = button.getBoundingClientRect();
-			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
-			const eventCatcher = container.querySelector( '.event-catcher' );
-			const eventCatcherRect = eventCatcher.getBoundingClientRect();
-			expect( buttonRect ).toEqual( eventCatcherRect );
-
-			await user.hover( eventCatcher );
-
-			// Tooltip hasn't appeared yet
-			expect(
-				screen.queryByText( 'Show helpful text here' )
-			).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
+			await user.hover( button );
 
-			// Tooltip shows after the delay
-			const tooltip = screen.getByText( 'Show helpful text here' );
-			expect( tooltip ).toBeVisible();
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
+			);
 
 			// Wait for the tooltip element to be positioned (aligned with the button)
 			await waitFor( () =>
 				expect(
-					getWrappingPopoverElement( tooltip )
+					getWrappingPopoverElement(
+						screen.getByText( 'tooltip text' )
+					)
 				).toBePositionedPopover()
 			);
 		} );
 
 		it( 'should not emit events back to children when they are disabled', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
-
+			const user = userEvent.setup();
 			const onClickMock = jest.fn();
+
 			const { container } = render(
-				<Tooltip text="Show helpful text here">
-					<button disabled onClick={ onClickMock }>
+				<Tooltip { ...props }>
+					<Button disabled onClick={ onClickMock }>
 						Click me
-					</button>
+					</Button>
 				</Tooltip>
 			);
 
+			// Note: this is testing for implementation details,
+			// but couldn't find a better way.
+			const buttonRect = screen
+				.getByRole( 'button', { name: 'Click me' } )
+				.getBoundingClientRect();
 			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
 			const eventCatcher = container.querySelector( '.event-catcher' );
 			await user.click( eventCatcher );
 			expect( onClickMock ).not.toHaveBeenCalled();
-		} );
 
-		it( 'should not show tooltip if the mouse leaves the anchor before the tooltip has shown', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
+			const eventCatcherRect = eventCatcher.getBoundingClientRect();
```

Comment:
Instead of creating another test [as we discussed](https://github.com/WordPress/gutenberg/pull/48397/files#r1121621557), I moved the implementation details to this test which is also testing the event catcher for a disabled button. Thoughts?

Reply:
Sounds good to me, thanks 👍